### PR TITLE
refine sn_hash

### PR DESCRIPTION
**Pull request by yufanho**
* Explicitly raising error when SN_HASH_KEY is not found to avoid silent data error
* Add common prefix to hash result so that user could easier distinguish values being/not being masked.
* Shorten hash result

### CODE REVIEW

File: plugin/trino-smartnews-functions/src/test/java/io/trino/plugin/smartnews/functions/TestHashFunctions.java [hunk 1]
```diff
@@ -72,11 +74,28 @@ public void testHashConsistency() throws Exception
...
+        String prevRes = runner.execute(session, "select sn_hash('123')").getOnlyValue().toString();
+        init();
+        assertNotSame(runner.execute(session, "select sn_hash('123')").getOnlyValue(), prevRes);
+    }
+
+    @Test
+    public void testHashingUTF8() throws Exception
+    {
+        String prevRes = runner.execute("select sn_hash('測試')").getOnlyValue().toString();
+        System.out.println(prevRes);
```
Comment (ubyyj) [inline plugin/trino-smartnews-functions/src/test/java/io/trino/plugin/smartnews/functions/TestHashFunctions.java]: I assume this output is not needed

File: plugin/trino-smartnews-functions/src/test/java/io/trino/plugin/smartnews/functions/TestHashFunctions.java [hunk 2]
```diff
@@ -72,11 +74,28 @@ public void testHashConsistency() throws Exception
     }
 
     @Test
-    public void testDynamics() throws Exception
+    public void testDynamicHashing() throws Exception
     {
-        String prevRes = runner.execute("select sn_hash('123')").getOnlyValue().toString();
+        Session session = TestingSession.testSessionBuilder().build();
+        String prevRes = runner.execute(session, "select sn_hash('123')").getOnlyValue().toString();
+        init();
```
Comment (ubyyj) [inline plugin/trino-smartnews-functions/src/test/java/io/trino/plugin/smartnews/functions/TestHashFunctions.java]: I don't know why, the init() could occasionally cause error: sn_hash is not registered.
So I guess you can initialized another session object, and use different session object to achieve your goal, without calling init() again.

File: plugin/trino-smartnews-functions/src/main/java/io/trino/plugin/smartnews/functions/HashFunctions.java [hunk 3]
```diff
@@ -29,15 +30,17 @@
...
 
 import static io.trino.spi.type.TypeUtils.readNativeValue;
 import static io.trino.spi.type.VarcharType.VARCHAR;
-import static java.nio.charset.StandardCharsets.UTF_8;
 
 public final class HashFunctions
 {
-    private static final String HASH_KEY = Optional.ofNullable(System.getenv("SN_HASH_KEY")).orElse("smartnews");
+    @SuppressWarnings("checkstyle:StaticVariableName")
+    private static String SN_HASH_KEY = System.getenv("SN_HASH_KEY");
```
Comment (ubyyj) [inline plugin/trino-smartnews-functions/src/main/java/io/trino/plugin/smartnews/functions/HashFunctions.java:40]: I am wondering can we just remove SN_HASH_KEY?
Since each query has different queryId, so the input to genSnHashSlice is dynamic already. What do you think?

Comment (yufanho) [reply plugin/trino-smartnews-functions/src/main/java/io/trino/plugin/smartnews/functions/HashFunctions.java:40]: Necessity of SN_HASH_KEY comes from the requirement of [irreproducibility](https://www.notion.so/Announcement-data-handling-policy-2nd-Party-Designated-PI-8200246de9a44f5a94ab38155a2c6307?pvs=4#bfe2362c68884f83ad248f8e284045b5), and some of discussion is [here](https://www.notion.so/Dynamic-Column-Masking-V2-4e6157cda7074848ba0df79eba928bb7?d=391cc20363274abbb13f6948969800df&pvs=4#447c9f932e5140fe92a64582c3b1692b).
Long story short, I think we can't remove it...

File: plugin/trino-smartnews-functions/src/test/java/io/trino/plugin/smartnews/functions/TestHashFunctions.java (hunk 2 above)
Comment (yufanho) [reply plugin/trino-smartnews-functions/src/test/java/io/trino/plugin/smartnews/functions/TestHashFunctions.java]: fixed

File: plugin/trino-smartnews-functions/src/test/java/io/trino/plugin/smartnews/functions/TestHashFunctions.java (hunk 1 above)
Comment (yufanho) [reply plugin/trino-smartnews-functions/src/test/java/io/trino/plugin/smartnews/functions/TestHashFunctions.java]: fixed

**Review by ubyyj — APPROVED** [review]
